File: Code_For_A_Cause/Noah_and_Marbles/test_case_generator.py
from random import randint
from math import gcd
def lcm(a,b):
    return (a*b)//(gcd(a,b))
def solve(a,n,x):
    ans=0
    for j in range(n):
        ans=gcd(ans,abs(a[j]-x))
    return ans
count_ones=0
input_file = open("input.txt","a")
output_file = open("output.txt","a")
t = 100
for test in range(t):
    n = 1000
    x = randint(-10**8,10**8)
    ans = randint(1,10**4)
    count = dict()
    input_file.write(str(n)+' '+str(x)+'\n')
    l=[]
    diff_list = []
    i = 0
    while i<n:
        multiple = randint(-10**4,10**4) * ans
        try:
            count[multiple]+=1
        except:
            count[multiple]=1
            diff_list.append(multiple)
            i+=1
    for i in range(n):
        coeff= randint(0,1)
        temp_values = [diff_list[i]+x,x-diff_list[i]]
        value = temp_values[coeff]
        l.append(value)
        if i==n-1:
            input_file.write(str(value)+'\n')
        else:
            input_file.write(str(value)+' ')
    answer = solve(l,n,x)
    output_file.write(str(answer)+'\n')

File: Code_For_A_Cause/The_Thanos_Challenge/test_case_generator.c
#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<math.h>
#include<ctype.h>
#include<limits.h>
#include<stdbool.h>
#include<time.h>
#include<assert.h>
#define and &&
#define or ||
#define not !
#define is ==
#define max(a,b) a>b?a:b
#define min(a,b) a<b?a:b
#define size 1000
#define mod ((int)(1e9+7))
typedef long long int ll;
int solve(int x, int y, int n, char *s)
{
    int ans = INT_MAX;
    if(x==0 && y==0)
        ans = 0;
	for(int i=0;i<n;i++)
    {
        if(s[i]=='S')
            y--;
        else if(s[i]=='N')
            y++;
        else if(s[i]=='E')
            x++;
        else if(s[i]=='W')
            x--;
		if(abs(x)+abs(y)<=i+1)
        {
            ans = min(ans, i+1);
            break;
        }
	}
    return ans;
}
int main()
{
    FILE *input,*output;
    srand(time(0));
    char directions[]="NEWS";
    input = fopen("input.txt","w");
    output = fopen("output.txt","w");
    int t = size;
    fprintf(input,"%d\n",t);
    int count = 0;
    for(int test = 0; test<t;test++)
    {
        int x, y, length;
        if(test<650)
        {
            x = rand()%250 + 1;
            y = rand()%250 + 1;
            length = rand()%400 + 1;
        }
        else
        {
            x = rand()%size + 1;
            y = rand()%size + 1;
            length = rand()%size + 1;
        }
        char * M = (char *)malloc(sizeof(char)*(length+1));
        if(test < 500)
        {
            for(int i=0;i<length; i++)
                M[i]= directions[rand()%3+1];
        }
        else
        {
            for(int i=0;i<length;i++)
                M[i]= directions[rand()%4];
        }
        M[length] ='\0';
        fprintf(input,"%d %d %s\n",x,y,M);
        int answer = solve(x,y,length,M);
        if(answer == INT_MAX)
        {
            fprintf(output,"IMPOSSIBLE\n");
            count++;
        }
        else
            fprintf(output,"%d\n",answer);
        free(M);
    }
    fclose(input);
    fclose(output);
    printf("%d",count);
    return 0;
}